README.md

```diff
@@ -31,6 +31,7 @@ First, make sure that you have the following dependencies:
   - Install [Node] (tested on v8.x.x).
   - Install [Yarn] (tested on v1.7.0).
   - Create a [GitHub API token]. No special permissions are required.
+  - Ensure that you have an SSH key (```id_rsa```) for ```git``` to use. On unix-like systems, this is usually stored in ```~/.ssh/```. If you do not, you can generate one by following these [instructions](https://help.github.com/en/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent).
```

Comment:
Hmm, @decentralion, didn’t we try to make it unnecessary to have SSH
keys unless you’re accessing a private repository? Did that regress?

(No action required for this change if you do actually lways need keys.)

Reply:
I probably induced this regression in https://github.com/sourcecred/sourcecred/pull/1085/files, specifically by changing our clone from https to git:
https://github.com/sourcecred/sourcecred/blob/4b2dcfa0ff87cde524b35398c8df12c822f4ec96/src/plugins/git/cloneAndLoadRepository.js#L19

Not sure what the best solution is. We should avoid needing extra steps for users to try out SourceCred, so fixing the bug will be important. but, it would be annoying to have an extra flag for cloning private repositories.

maybe we could have it try cloning with https, and then if it fails re-try with git, and print appropriate error messages depending on what happens? or too hacky?

Reply:
If I had to opt for one option over the other, I'd go with flags over keys every time. As it stands right now, too, we are forcing key gen (or at least a check for some people) for everyone no matter the type of repo they are running SourceCred against. A flag might be an extra step but it's an extra step for a smaller number of people (only those who need to run against a private repo).

Reply:
Yep, I agree. I wrote a detailed description along with an implementation plan here: https://github.com/sourcecred/sourcecred/issues/1168. As a very first step, we can just swap back from git to https; AFAIK no one except me has ever used the private git loading feature, so I'm fine privileging the common case over the exceptional case. @tylerdmace would you be willing to take this on? (either Just Fixing it hackily, or adding the flag.)

For now, let's keep this change to the README (since it is accurate information for SourceCred as it presently exists) and let's plan to take it back out once we fix the underlying issue.

Reply:
Absolutely, I can take it!